zipline/sources/test_source.py

```diff
@@ -88,7 +87,7 @@ def advance_current(cur):
 
     # yield count trade events, all on trading days, and
     # during trading hours.
-    for i in range(count):
+    while cur < end:
```

Comment:
Yes, and I'm just now seeing that, in the case of default args, this would mean no dates. Should I remove the default args?